### ISSUE TITLE
File.readable? returns incorrect result if file does not exist

### ISSUE THREAD

**Issue by cnjmike** [fakefs/fakefs#419]
To recreate

```
no_file_at = '/bogus/path'
puts File.readable?(no_file_at)  # => false

require 'fakefs'
File.readable?(no_file_at) # => Error::ENOENT
```

This is breaking all our tests that use both FakeFS and Net::SSH (which tests for existence and readability of optional default config files in one go with just File.readable?)

**Comment by gnikonorov** [comment]
Thank you for raising this issue @cnjmike! I’ll take a look at this tonight and fix it.

**Comment by grosser** [comment]
0.18.0

**Comment by gnikonorov** [comment]
Thank you @grosser!
@cnjmike - can you please confirm proper functionality?

**Comment by gnikonorov** [comment]
Closing as this issue is now fixed as of gem version `0.18.0`
@cnjmike - please feel free to re-open this issue if the problem is not solved.
